app/models/seance.rb

```diff
@@ -0,0 +1,28 @@
+class Seance < ApplicationRecord
+  belongs_to :hall
+  belongs_to :movie
+
+  before_validation :end_time
+
+  def end_time
+    self.end_time = self.start_time + self.movie.duration.minutes + 30.minutes
+  end
```

Comment:
```suggestion
  before_validation :set_end_time

  def set_end_time
    end_time = start_time + movie.duration.minutes + 30.minutes
  end
```

Having the method be the same as the column might cause issues but you won't see that immediately. The reason is that the assignment on line 8 will return its value so your `is_used` method will work correctly but it will cause an N+1 issue as you will always be fetching the movie record. 
You also don't need to use `self` in here - self is implied by default in this scope :) 

You can read about it here: 
[N+1 problem](https://semaphoreci.com/blog/2017/08/09/faster-rails-eliminating-n-plus-one-queries.html)
[Self in Ruby](https://www.rubyguides.com/2020/04/self-in-ruby/)

Reply:
Thanks for all the comments! Regarding this particular comment i have a question. Is the self in "self.end_time" not needed? Without it after the method ends the "end_time" value return to being nil while with it It keeps its value.

Reply:
I just checked that and you're right! That's the only self that needs to stay 👍🏻